crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -503,8 +506,24 @@ impl KitsuneP2pEventHandler for KitsuneP2pActor {
     fn handle_put_agent_info_signed(
         &mut self,
         input: crate::event::PutAgentInfoSignedEvt,
-    ) -> KitsuneP2pEventHandlerResult<()> {
-        Ok(self.host_api.legacy.put_agent_info_signed(input))
+    ) -> KitsuneP2pEventHandlerResult<Vec<AgentInfoPut>> {
+        let legacy_host = self.host_api.legacy.clone();
+        let ep_hnd = self.ep_hnd.clone();
+
+        Ok(async move {
+            let puts = legacy_host.put_agent_info_signed(input).await?;
+
+            for removed_url in puts.iter().flat_map(|r| r.removed_urls.clone()) {
+                tracing::debug!(?removed_url, "peer URL changed, closing connection");
+                if let Err(e) = ep_hnd.close_peer_con(removed_url.clone()) {
+                    tracing::info!(?e, ?removed_url, "could not close peer connection");
```

Comment:
Can this happen in any "normal behavior" circumstances... if so, maybe this should be `debug!` level. Mostly worried about things like `e='connection already closed' could not close peer connection`. Or we can leave this as-is, and see if anything weird starts getting printed : )

Reply:
Yes this can happen, if both conductors restart between connections to each other. E.g. if they both shut down until agent info expires. Then there won't be an open connection to close when seeing the peer URL change. Changed to debug